test_journal.py

```diff
@@ -411,3 +413,88 @@ def test_bdd_test_for_color(app, color_entry):  # Get a test entry
     assert FUNC_NAME in response
+
+
+# Pytest-splinter tests require app running on localhost:5000
+
+def splinter_login_helper(browser):
+    browser.visit('http://127.0.0.1:5000/login')
+    browser.fill('username', 'admin')
+    browser.fill('password', 'secret')
+    browser.find_by_value('Login')[0].click()
+
+
+# Add ajax
+
+def test_create_response_url():
+    with Browser() as browser:
+        splinter_login_helper(browser)
+        browser.visit('http://127.0.0.1:5000/create')
+        browser.fill('title', 'A title')
+        browser.fill('content', 'Some content')
+        browser.find_by_value('Share')[0].click()
+        assert browser.url == 'http://127.0.0.1:5000/create'
+
+
+def test_edit_response_url():
+    with Browser() as browser:
+        splinter_login_helper(browser)
+        browser.visit('http://127.0.0.1:5000/')
+        browser.find_by_value('View')[0].click()
+        browser.find_by_value('Edit')[0].click()
+        browser.fill('title', 'changed')
+        browser.fill('content', 'changed')
+        eid = browser.find_by_tag('input')[1].value
+        browser.find_by_value('Commit')[0].click()
+        assert browser.url == 'http://127.0.0.1:5000/edit?id=%s&edit=Edit' % eid
+
+
+# Add functioning without javascript
+
+def no_js(browser):
+    browser.visit('chrome://settings-frame/content')
+    browser.find_by_name('javascript')[1].click()
+    browser.find_by_id('content-settings-overlay-confirm').click()
+
+
+def restore_js(browser):
+    browser.visit('chrome://settings-frame/content')
+    browser.find_by_name('javascript')[0].click()
+    browser.find_by_id('content-settings-overlay-confirm').click()
+
+
+def test_create_without_javascript():
+    with Browser('chrome') as browser:
+        no_js(browser)
+        splinter_login_helper(browser)
+        time.sleep(0.1)
+        browser.visit('http://127.0.0.1:5000/create')
+        time.sleep(0.1)
+        browser.fill('title', 'Another title')
+        browser.fill('content', 'Different content')
+        browser.find_by_value('Share')[0].click()
+        time.sleep(0.1)
+        assert browser.url == 'http://127.0.0.1:5000/'
+        created = browser.find_by_tag('h2')[0].text
+        assert created == 'Another title'
+        restore_js(browser)
+
+
+def test_edit_without_javascript():
+    with Browser('chrome') as browser:
+        no_js(browser)
+        splinter_login_helper(browser)
+        time.sleep(0.1)
+        browser.visit('http://127.0.0.1:5000/')
+        browser.find_by_value('View')[0].click()
+        time.sleep(0.1)
+        browser.find_by_value('Edit')[0].click()
+        time.sleep(0.1)
+        browser.fill('title', 'new change')
+        browser.fill('content', 'new change')
+        browser.find_by_value('Commit')[0].click()
+        time.sleep(0.1)
+        assert browser.url == 'http://127.0.0.1:5000/'
+        changed = browser.find_by_tag('h2')[0].text
+        assert changed == 'new change'
+        restore_js(browser)
```

Comment:
I really like that you've implemented tests that prove that your editing works whether you've got js enabled or not.  Good work!